test/auto_parallel/hybrid_strategy/semi_auto_parallel_llama_model.py

```diff
@@ -684,34 +661,25 @@ def forward(
 
             has_gradient = not hidden_states.stop_gradient
             ipp = decoder_layer.ipp
-
-            if ipp is not None and pre_ipp != ipp:
-                if (
-                    not paddle.in_dynamic_mode()
-                    and getattr(self.config, "virtual_pp_degree", 1) > 1
-                ):
-                    hidden_states = dist.reshard(
-                        hidden_states,
-                        get_mesh(ipp),
-                        self.placements,
-                    )
-                    decoder_layer = dist.shard_op(decoder_layer, get_mesh(ipp))
-                else:
-                    hidden_states = dist.reshard(
-                        hidden_states,
-                        get_mesh(ipp),
-                        self.placements,
-                    )
-                    position_ids = dist.reshard(
-                        position_ids,
-                        get_mesh(ipp),
-                        [dist.Shard(0), dist.Replicate()],
-                    )
-                    attention_mask = dist.reshard(
-                        attention_mask,
-                        get_mesh(ipp),
-                        [dist.Shard(0), dist.Replicate()],
-                    )
+            if ipp is None:  # without pp
```

Comment:
the idx that needs reshard "pp_stage*[1,2,3, num_layers/pp_degree -1 ]"

Reply:
Done